python/desc/sims/GCRCatSimInterface/CatalogClasses.py

```diff
@@ -4,15 +4,91 @@
 from lsst.utils import getPackageDir
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSersic2D
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogZPoint
+from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSN
 from lsst.sims.catUtils.mixins import VariabilityAGN
 from lsst.sims.catalogs.decorators import cached, compound
 from lsst.sims.catUtils.mixins import EBVmixin
 
-__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN"]
+from lsst.sims.catalogs.db import fileDBObject
+
+__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN", "SNFileDBObject",
+           "DC2PhosimCatalogSN"]
 
 #########################################################################
 # define a class to write the PhoSim catalog; defining necessary defaults
 
+class SNFileDBObject(fileDBObject):
+    """
+    Use FileDBObject to provide CatalogDBObject functionality for SN
+    with host galaxies from protoDC2 output to csv files before
+    """
+    dbDefaultValues = {'varsimobjid':-1,
+                       'runid':-1,
+                       'ismultiple':-1,
+                       'run':-1,
+                       'runobjid':-1}
+
+    # These types should be matched to the database.
+    #: Default map is float.  If the column mapping is the same as the
+    # column name, None can be specified
+
+    columns = [('raJ2000', 'snra_in*PI()/180.'),
+               ('decJ2000', 'sndec_in*PI()/180.'),
+               ('Tt0', 't0_in'),
+               ('Tx0', 'x0_in'),
+               ('Tx1', 'x1_in'),
+               ('Tc', 'c_in'),
+               ('id', 'snid_in'),
+               ('Tredshift', 'z_in'),
+               ('redshift', 'z_in'),
+               ('Tgaltileid', 'galtileid')
+              ]
+class DC2PhosimCatalogSN(PhoSimCatalogSN):
+    """
+    Modification of the PhoSimCatalogSN mixin to provide shorter sedFileNames
+    by leaving out the parts of the directory name. Also fix name changes from
+    gamma to shear.
+    """
+    def get_sedFilepath(self):
```

Comment:
Do we really need to map `TsedFilepath` into `sedFilepath`, given that the catalog is just going to output `shorterFileNames`, anyway?  Shouldn't `get_shorterFileNames` just call `self.column_by_name('TsedFilepath')`?

Reply:
Don't know. Not planning to change something that is working at this point.

Reply:
OK.  We can look at this as we refactor things for Run 2.0